realm/realm-library/src/androidTest/kotlin/io/realm/entities/embedded/EmbeddedSimpleListParent.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.entities.embedded
+
+import io.realm.RealmList
+import io.realm.RealmObject
+import io.realm.annotations.PrimaryKey
+
+// Top-level object describing a simple embedded objects structure consisting of only a
+// list of embedded objects.
+open class EmbeddedSimpleListParent : RealmObject {
+    constructor() {}
+    constructor(id: String) : super() {
+        this.id = id
+    }
+
+    @PrimaryKey
+    var id: String? = null
+    var children: RealmList<EmbeddedSimpleChild> = RealmList()
```

Comment:
We don't support final fields in the annotation processor. Probably mostly an oversight which is a lot more annoying in Kotlin. I created https://github.com/realm/realm-java/issues/6892. I suspect it would be a fairly trivial fix.

Reply:
Good to know, I had no idea.